### PR TITLE
Use different muparserx library in linux depending on the ABI

### PR DESCRIPTION
**Pull request by vvilpas**
### Summary

Solves problems related to muparserx when using CENTOS 7, as reported by John Smolin.

### Details and comments
The problem comes from the impossibility to use the new C++11_ABI for gcc compilers running under RHEL6 & RHEL7 ( and every linux distro based on those ones, like centos). Another version of the muparserx version compiled with the old ABI is provided in order to solve this issue.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -200,16 +200,25 @@ else()
...
-	set(PLATFORM "win64")
-elseif(APPLE)
-	set(PLATFORM "macos")
-elseif(UNIX)
-	set(PLATFORM "linux")
-endif()
-execute_process(COMMAND ${CMAKE_COMMAND} -E tar "xvfj" "${AER_SIMULATOR_CPP_SRC_DIR}/third-party/${PLATFORM}/lib/muparserx.7z"
-	WORKING_DIRECTORY  "${AER_SIMULATOR_CPP_SRC_DIR}/third-party/${PLATFORM}/lib/")
-set(MUPARSERX_LIB_PATH "${AER_SIMULATOR_CPP_SRC_DIR}/third-party/${PLATFORM}/lib")
+#if(MSVC)
```
Comment (atilag) [inline CMakeLists.txt]: Ok, just remove all these commented lines. We can always go back in git history to see this code.

File: cmake/compiler_utils.cmake [hunk 2]
```diff
@@ -56,4 +56,31 @@ function(get_muparserx_source_code)
                 Please download MuparserX library from https://github.com/beltoforion/muparserx.git \
                 and checkout latest stable release")
     endif()
-endfunction()
\ No newline at end of file
+endfunction()
+
+function(check_compiler_cpp11_abi)
+    # Consider also if -D_GLIBCXX_USE_CXX11_ABI has been passed as flag
```
Comment (atilag) [inline cmake/compiler_utils.cmake]: Can we comment here why all this is needed? With the link to the original issue (bugzilla) and a short description... Our "old me" and newcomers to the project will thank some more clarification :)

File: cmake/compiler_utils.cmake [hunk 3]
```diff
@@ -56,4 +56,31 @@ function(get_muparserx_source_code)
...
+    string(REGEX MATCH "-D_GLIBCXX_USE_CXX11_ABI=[(A-z)|(a-z)|(0-9)]+" CUSTOM_PREP_FLAGS ${CMAKE_CXX_FLAGS})
+    # Preprocessor run to check if CXX11_ABI is set
+    execute_process(COMMAND echo "#include <string>" COMMAND ${CMAKE_CXX_COMPILER} ${CUSTOM_PREP_FLAGS} -x c++ -E -dM -  COMMAND fgrep _GLIBCXX_USE_CXX11_ABI OUTPUT_VARIABLE C++11_ABI_OUT OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REGEX REPLACE "#define _GLIBCXX_USE_CXX11_ABI " "" C++11_ABI "${C++11_ABI_OUT}")
+    set(C++11_ABI ${C++11_ABI} PARENT_SCOPE)
```
Comment (atilag) [inline cmake/compiler_utils.cmake]: I'm surprised that cmake allows '+' as a variable name...

File: CMakeLists.txt (hunk 1 above)
Comment (vvilpas) [reply CMakeLists.txt]: Yeah! I forgot!

File: cmake/compiler_utils.cmake (hunk 2 above)
Comment (vvilpas) [reply cmake/compiler_utils.cmake]: Totally agree!

File: cmake/compiler_utils.cmake (hunk 3 above)
Comment (vvilpas) [reply cmake/compiler_utils.cmake]: Actually, I've seen problems with '+' in names so I'll change them.

**Pull request merged**